src/main/java/net/atos/entng/rbs/filters/TypeAndResourceAppendPolicy.java

```diff
@@ -155,6 +155,108 @@ public void authorize(final HttpServerRequest request, final Binding binding, fi
 		}
 	}
 
+	public void authorize(String resourceId, String bookingId, final Binding binding, final UserInfos user, final Handler<Boolean> handler) {
+		SqlConf conf = SqlConfs.getConf(ResourceController.class.getName());
+		boolean hasBooking = (bookingId != null && !bookingId.trim().isEmpty());
+
+		if (resourceId != null && !resourceId.trim().isEmpty() && (parseId(resourceId) instanceof Integer)) {
+			// Replace character '.' by '-' in the called method's name
+			String sharedMethod = binding.getServiceMethod().replaceAll("\\.", "-");
+
+			// Groups and users
+			final List<String> groupsAndUserIds = new ArrayList<>();
+			groupsAndUserIds.add(user.getUserId());
+			if (user.getGroupsIds() != null) {
+				groupsAndUserIds.addAll(user.getGroupsIds());
+			}
+
+			// Query
+			StringBuilder query = new StringBuilder();
+			JsonArray values = new fr.wseduc.webutils.collections.JsonArray();
+			query.append("SELECT count(*)")
+					.append(" FROM rbs.resource AS r")
+					.append(" INNER JOIN rbs.resource_type AS t ON r.type_id = t.id");
+			if (hasBooking) {
+				// Additional join when parameter bookingId is used
+				query.append(" INNER JOIN rbs.booking AS b ON b.resource_id = r.id");
+			}
+
+			query.append(" LEFT JOIN rbs.resource_type_shares AS ts ON t.id = ts.resource_id")
+					.append(" LEFT JOIN rbs.resource_shares AS rs ON r.id = rs.resource_id")
+					.append(" WHERE (");
+
+			if (isUpdateBooking(binding) || isUpdatePeriodicBooking(binding) || isDeleteBooking(binding)) {
+				query.append("(ts.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND ts.action = 'net-atos-entng-rbs-controllers-BookingController|processBooking')");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
+
+				query.append(" OR (rs.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND rs.action = 'net-atos-entng-rbs-controllers-BookingController|processBooking')");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
+			} else {
+				query.append("(ts.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND ts.action = ?)");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
+				values.add(sharedMethod);
+
+				query.append(" OR (rs.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND rs.action = ?)");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
```

Comment:
values.addAll(new JsonArray(groupsAndUserIds))

Reply:
authorize duplicate with arguments so that no HTTPRequest is needed (polymorph)